socket_linux.go

```diff
@@ -18,11 +18,11 @@ func NewSocket(addr *net.UDPAddr, recvBuf int, reuseport bool) (net.PacketConn,
 	// unix.SO_REUSEPORT is not defined on linux 386/amd64, see
 	// https://github.com/golang/go/issues/16075
 	if reuseport {
-		if err := unix.SetsockoptInt(sockFD, unix.SOL_SOCKET, 0xf, 1); err != nil {
+		if err = unix.SetsockoptInt(sockFD, unix.SOL_SOCKET, 0xf, 1); err != nil {
```

Comment:
Was this actually a golint error? I'm not getting it locally, and furthermore, I'm not sure that this fix is actually what we want.